backend/services/latex_service.py

```
@@ -68,37 +68,56 @@ async def process_latex_file(
-                    "-halt-on-error",
-                    f"{safe_filename}.tex",
-                ],
-                cwd=temp_dir,
-                env=tex_environment,
-                capture_output=True,
-                text=True,
-                timeout=30,
-            )
+            result = None
+            for _pass_number in range(2):
+                result = subprocess.run(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)